lib/msal-browser/docs/acquire-token.md

```diff
@@ -89,10 +79,89 @@ var graphEndpoint = "https://graph.microsoft.com/v1.0/me";
 
 fetch(graphEndpoint, options)
     .then(resp => {
-            //do something with response
+        //do something with response
     });
 ```
 
+## MSAL token acquisition best practices
+
+The following are best practices to acquire tokens with MSAL for avoiding errors, performance hits and usability issues. Certain scenarios may provide exceptions to these.
+
+### Use a single PublicClientApplication instance
+
+Instantiate one `PublicClientApplication` per application and use the same instance throughout your app. This ensures that there is a single source of truth for what MSAL is performing at any given time (see: [MSAL events](events.md)) and eliminates the chance of distinct app objects making parallel interactive requests or potential cache conflicts, which might break apps, reduce performance or hinder user experience.
+
+### Always wait for promises to resolve
+
+All MSAL `acquireToken*` as well as `login*` APIs perform asynchronous operations and return promises. You should always wait for these promises to resolve, even if you do not need the promise payload.
```

Comment:
Let's clarify that you should wait for promises to resolve before doing any other tasks that depend on authentication state or tokens such as rendering user information, calling a protected API or calling other MSAL APIs. We don't want to give the impression that these async functions should in fact be synchronous.

Reply:
Good point -revised this bit